plugins/woocommerce-blocks/tests/e2e/tests/product-collection/product-collection.block_theme.spec.ts

```diff
@@ -961,3 +968,253 @@ test.describe( 'Product Collection', () => {
+				} );
+			}
+
+			test( `Collection "${ collection.name }" should not show preview label in a post`, async ( {
+				pageObject,
+				editor,
+				admin,
+			} ) => {
+				await admin.createNewPost();
+				await pageObject.insertProductCollection();
+				await pageObject.chooseCollectionInPost( key as Collections );
+
+				// Check visibility of product picker
+				const editorProductPicker = admin.page.locator(
+					SELECTORS.productPicker
+				);
+				const expectedVisibility = collection.shouldShowProductPicker
+					? 'toBeVisible'
+					: 'toBeHidden';
+				await expect( editorProductPicker )[ expectedVisibility ]();
+
+				if ( collection.shouldShowProductPicker ) {
+					await pageObject.chooseProductInEditorProductPickerIfAvailable(
+						admin.page
+					);
+				}
+
+				// At this point, the product picker should be hidden
+				await expect( editorProductPicker ).toBeHidden();
+
+				// Check visibility of preview label
+				const block = editor.canvas.getByLabel( collection.label );
+				const previewButtonLocator = block.getByTestId(
+					SELECTORS.previewButtonTestID
+				);
+
+				await expect( previewButtonLocator ).toBeHidden();
+			} );
+
+			test( `Collection "${ collection.name }" should not show preview label in Product Catalog template`, async ( {
+				pageObject,
+				editor,
+			} ) => {
+				await pageObject.goToProductCatalogAndInsertCollection(
+					key as Collections
+				);
+
+				const block = editor.canvas.getByLabel( collection.label );
+				const previewButtonLocator = block.getByTestId(
+					SELECTORS.previewButtonTestID
+				);
+
+				await expect( previewButtonLocator ).toBeHidden();
+			} );
+		}
+	);
+} );
+
+test.describe( 'Editor product picker', () => {
+	const MY_REGISTERED_COLLECTIONS_THAT_NEEDS_PRODUCT = {
+		myCustomCollectionWithProductContext: {
+			name: 'My Custom Collection - Product Context',
+			label: 'Block: My Custom Collection - Product Context',
+			collection:
+				'woocommerce/product-collection/my-custom-collection-product-context',
+		},
+		myCustomCollectionMultipleContexts: {
+			name: 'My Custom Collection - Multiple Contexts',
+			label: 'Block: My Custom Collection - Multiple Contexts',
+			collection:
+				'woocommerce/product-collection/my-custom-collection-multiple-contexts',
+		},
+	};
+
+	// Activate plugin which registers custom product collections
+	test.beforeEach( async ( { requestUtils } ) => {
+		await requestUtils.activatePlugin(
+			'register-product-collection-tester'
+		);
+	} );
+
+	Object.entries( MY_REGISTERED_COLLECTIONS_THAT_NEEDS_PRODUCT ).forEach(
+		( [ key, collection ] ) => {
+			test( `For collection "${ collection.name }" - manually selected product reference should be available on Frontend in a post`, async ( {
+				pageObject,
+				admin,
+				page,
+			} ) => {
+				await admin.createNewPost();
+				await pageObject.insertProductCollection();
+				await pageObject.chooseCollectionInPost( key as Collections );
```

Comment:
Can be replaced with:

```suggestion
				await pageObject.createNewPostAndInsertBlock( key as Collections );
```

Reply:
As I explained [here](https://github.com/woocommerce/woocommerce/pull/50590#discussion_r1738359872), I can't use it because `createNewPostAndInsertBlock` also select a product by calling `chooseProductInEditorProductPickerIfAvailable` function. For this test, I wanna call it manually. Does it make sense?